rdt/hyper_transformer.py

```diff
@@ -300,7 +304,61 @@ def set_config(self, config):
         if self._fitted:
             warnings.warn(self._REFIT_MESSAGE)
 
-    def update_transformers_by_sdtype(self, sdtype, transformer):
+    def _validate_update_transformers_by_sdtype(
+            self, sdtype, transformer, transformer_name, transformer_parameters):
+        if not self.field_sdtypes:
+            raise Error(
+                'Nothing to update. Use the `detect_initial_config` method to '
+                'pre-populate all the sdtypes and transformers from your dataset.'
+            )
+
+        if transformer_name is None:
+            if transformer is None:
+                raise ValueError("Missing required parameter 'transformer_name'.")
+
+            if not isinstance(transformer, BaseTransformer):
+                raise Error('Invalid transformer. Please input an rdt transformer object.')
+
+            if sdtype not in transformer.get_supported_sdtypes():
+                raise Error("The transformer you've assigned is incompatible with the sdtype.")
+
+        else:
+            if transformer_name not in self._TRANSFORMERS_TO_CLASS or sdtype not in \
+                    self._TRANSFORMERS_TO_CLASS[transformer_name].get_supported_sdtypes():
+                raise ValueError(
+                    f"Invalid transformer name '{transformer_name}' for the '{sdtype}' sdtype.")
+
+            if transformer_parameters is not None:
+                transformer = self._TRANSFORMERS_TO_CLASS[transformer_name]
+                valid = inspect.signature(transformer).parameters
+                invalid_parameters = {arg for arg in transformer_parameters if arg not in valid}
+                if invalid_parameters:
+                    raise ValueError(
+                        f'Invalid parameters {tuple(sorted(invalid_parameters))} '
+                        f"for the '{transformer_name}'."
+                    )
+
+    def _warnings_update_transformers_by_sdtype(self, transformer, transformer_name):
```

Comment:
minor: I think we should try to make all methods verbs. so `_check_warnings...` or `_warn..`